db/sqlc/migrations/000001_accounts.up.sql

```diff
@@ -0,0 +1,71 @@
+CREATE TABLE IF NOT EXISTS accounts (
+    -- The auto incrementing primary key.
+    id INTEGER PRIMARY KEY,
+
+    -- The ID that was used to identify the account in the legacy KVDB store.
+    -- In order to avoid breaking the API, we keep this field here so that
+    -- we can still look up accounts by this ID for the time being.
+    alias BLOB NOT NULL UNIQUE,
+
+    -- An optional label to use for the account. If it is set, it must be
+    -- unique.
+    label TEXT UNIQUE,
+
+    -- The account type.
+    type SMALLINT NOT NULL,
+
+    -- The accounts initial balance. This is never updated.
+    intial_balance_msat BIGINT NOT NULL,
+
+    -- The accounts current balance. This is updated as the account is used.
+    current_balance_msat BIGINT NOT NULL,
+
+    -- The last time the account was updated.
+    last_updated TIMESTAMP NOT NULL,
+
+    -- The time that the account will expire.
+    expiration TIMESTAMP NOT NULL
+);
+
+-- The account_payments table stores all the payment hashes of outgoing
+-- payments that are associated with a particular account. These are used to
+-- when an account should be debited.
+CREATE TABLE IF NOT EXISTS account_payments (
+    -- The account that this payment is linked to.
+    account_id BIGINT NOT NULL REFERENCES accounts(id) ON DELETE CASCADE,
+
+    -- The payment hash of the payment.
+    hash BLOB NOT NULL,
+
+    -- The LND RPC status of the payment.
+    status SMALLINT NOT NULL,
+
+    -- The total amount of the payment in millisatoshis.
+    -- This includes the payment amount and estimated routing fee.
+    full_amount_msat BIGINT NOT NULL,
+
+    UNIQUE(account_id, hash)
+);
+
+-- The account_invoices table stores all the invoice payment hashes that
+-- are associated with a particular account. These are used to determine
+-- when an account should be credited.
+CREATE TABLE IF NOT EXISTS account_invoices (
+    -- The account that this invoice is linked to.
+    account_id BIGINT NOT NULL REFERENCES accounts(id) ON DELETE CASCADE,
+
+    -- The payment hash of the invoice.
+    hash BLOB NOT NULL,
+
+    UNIQUE(account_id, hash)
```

Comment:
these unique constraints are only for the individual tables, right? i.e. what happens if the same account tries to pay itself (invoice+payment), will it violate the constraint (if there would be only a single index)?

Reply:
yeah these are per table